xds/src/main/java/io/grpc/xds/Bootstrapper.java

```diff
@@ -74,42 +79,47 @@ public static Bootstrapper getInstance() {
   /**
    * Returns configurations from bootstrap.
    */
-  public abstract BootstrapInfo readBootstrap() throws IOException;
+  public abstract BootstrapInfo readBootstrap() throws XdsInitializationException;
 
   /** Parses a raw string into {@link BootstrapInfo}. */
   @VisibleForTesting
-  @SuppressWarnings("deprecation")
-  public static BootstrapInfo parseConfig(String rawData) throws IOException {
+  @SuppressWarnings("unchecked")
+  public static BootstrapInfo parseConfig(String rawData) throws XdsInitializationException {
```

Comment:
A general comment for `parseConfig`: wouldn't it be better to do strictly parsing and schema checks here and do any semantic checks later either as a separate step or where the `BootstrapInfo` field is consumed? This applies to checks for missing server_uri (this is a gray area I admit), channel creds type etc. That will be cleaner.

Reply:
It is doing checks in this way, the parser does not look at what channel creds types are provided, it's just parsing them to Strings and let `BootstrapInfo` consumers handle if anything they are looking for is missing. Empty server_uri is an ambiguous one, as from the parser's point of view, you could say if server_uri is an empty JSON it would fail to create the `BootstrapInfo` object (the object does not make sense to exist) or you could say just give `BootstrapInfo` consumers an empty list and let them handle it. Both are acceptable to me, but it would be good to do this consistently for both server_uri and channel_creds unless there is a special requirement.